CIScripts/Tests/SNATTest.ps1

```diff
@@ -1,3 +1,350 @@
+    New-SNATVM -Session $Session `
+        -VmDirectory $SNATVMDir -DiskPath $SNATDiskPath `
+        -MgmtSwitchName $SNATMgmtSwitchName `
+        -VRouterSwitchName $TestConfiguration.VMSwitchName `
+        -RightGW $SNATConfiguration.GatewayIP `
+        -LeftGW $SNATConfiguration.ContainerGatewayIP `
+        -ForwardingMAC $SNATVeth.MacAddress `
+        -GUID $SNAT_GUID
+
+    Write-Host "Extracting adapters data..."
+    $PhysicalAdapter = Get-RemoteNetAdapterInformation -Session $Session -AdapterName $TestConfiguration.AdapterName
+    $HNSAdapter = Get-RemoteNetAdapterInformation -Session $Session -AdapterName HNSTransparent
+    $SNATLeft = Get-RemoteVMNetAdapterInformation -Session $Session -VMName $SNATVMName -AdapterName $SNATLeftName
+    $SNATRight = Get-RemoteVMNetAdapterInformation -Session $Session -VMName $SNATVMName -AdapterName $SNATRightName
+    Write-Host "Extracting adapters data... DONE"
+
+    $SNATLeftVif = 102
+    $SNATLeftNh = 102
+    $SNATRightVif = 103
+    $SNATRightNh = 103
+
+    Write-Host "Setting up routing rules..."
+    $Res = Invoke-Command -Session $Session -ScriptBlock {
+        route add $Using:SNATConfiguration.GatewayIP mask 255.255.255.255 $Using:SNATConfiguration.VethIP "if" $Using:SNATVeth.IfIndex | Out-Null
+        $LASTEXITCODE
+    }
+    if ($Res -ne 0) {
+        throw "Setting up routing rules... FAILED"
+    }
+    Write-Host "Setting up routing rules... DONE"
+
+    Write-Host "Setting up ARP rule for GW-veth forwarding..."
+    $Res = Invoke-Command -Session $Session -ScriptBlock {
+        netsh interface ipv4 add neighbor $Using:SNATVeth.IfIndex $Using:SNATConfiguration.GatewayIP $Using:SnatRight.MacAddressWindows | Out-Null
+        $LASTEXITCODE
+    }
+    if ($Res -ne 0) {
+        throw "Setting up ARP rule for GW-veth forwarding... FAILED"
+    }
+    Write-Host "Setting up ARP rule for GW-veth forwarding... DONE"
+
+    Write-Host "Enable forwarding on the HNSTransparent adapter..."
+    $Res = Invoke-Command -Session $Session -ScriptBlock {
+        netsh interface ipv4 set interface $Using:HNSAdapter.IfIndex forwarding="enabled" | Out-Null
+        $LASTEXITCODE
+    }
+    if ($Res -ne 0) {
+        throw "Enable forwarding on the HNSTransparent adapter... FAILED"
+    }
+    Write-Host "Enable forwarding on the HNSTransparent adapter... DONE"
+
+    Write-Host "Start and configure test container..."
+    $DockerNetwork = $TestConfiguration.DockerDriverConfiguration.NetworkConfiguration.NetworkName
+    $ContainerID = Invoke-Command -Session $Session -ScriptBlock { docker run -id --network $Using:DockerNetwork microsoft/nanoserver powershell }
+    $ContainerNetInfo = Get-RemoteContainerNetAdapterInformation -Session $Session -ContainerID $ContainerID
+    $ContainerVIF = 101
+    $ContainerNH = 101
+    Write-Host "Start and configure test container... DONE"
+
+    $InternalVrf = 1
+    $ExternalVrf = 2
+
+    $BroadcastInternalNh = 110
+    $BroadcastExternalNh = 111
```

Comment:
Lots of hardcoded values here (and above), but I'm kinda torn if they should be moved to some external struct or stay here. 

In any case, definitions of `ContainerVIF/NH`, `SNATLeftVif` and these are kinda all over the code right now...